src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/PoPUserOTPServiceTest.kt

```diff
@@ -66,7 +66,9 @@ class PoPUserOTPServiceTest {
 
     Mockito.`when`(popUserOTPRepository.findByPrisoner(any())).thenReturn(popUserOTPEntity)
     val result = popUserOTPService.getOTPByPrisoner(prisonerEntity)
-    Assertions.assertEquals(popUserOTPEntity, result)
+    if (result != null) {
```

Comment:
if result is null the test will pass. Would be better to remove the if and do
`Assertions.assertEquals(popUserOTPEntity.id, result!!.id)`

Reply:
removed the if